internal/generator/crypto/hash/mimc/template/tests/mimc_test.go.tmpl

```diff
@@ -0,0 +1,114 @@
+import (
+	"bytes"
+	"testing"
+
+	"github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr"
+	"github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr/mimc"
+	fiatshamir "github.com/consensys/gnark-crypto/fiat-shamir"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMiMCFiatShamir(t *testing.T) {
+	fs := fiatshamir.NewTranscript(mimc.NewMiMC(), "c0")
+	zero := make([]byte, mimc.BlockSize)
+	err := fs.Bind("c0", zero)
+	assert.NoError(t, err)
+	_, err = fs.ComputeChallenge("c0")
+	assert.NoError(t, err)
+}
+
+func TestByteOrder(t *testing.T) {
+	assert := require.New(t)
+
+	var buf [fr.Bytes]byte
+	// if the 31 first bytes are FF, it's a valid FF in little endian, but not in big endian
```

Comment:
I guess here "31" should be {{ sub .Fr.Bytes 1 }} or something like that (unrelated to this PR I know :) )